streams/src/main/java/org/apache/kafka/streams/kstream/internals/MaterializedInternal.java

```diff
@@ -40,6 +40,11 @@ public MaterializedInternal(final Materialized<K, V, S> materialized,
         }
     }
 
+    public MaterializedInternal(Materialized<K, V, S> materialized) {
+        super(materialized);
+        queryable = storeName() != null;
```

Comment:
It's dangerous to call overridable instance methods during construction. The recommendation is to make `storeName()` final, static, or private, or avoid it in constructors. This applies to the other constructor as well.

Reply:
Ack, this will be taken out in the refactoring for comments.